src/xwayland/Server.cpp

```diff
@@ -21,124 +18,156 @@
+    socklen_t size = offsetof(struct sockaddr_un, sun_path) + path_size + 1;
+    int       fd = socket(AF_UNIX, SOCK_STREAM, 0);
     if (fd < 0) {
-        Debug::log(ERR, "failed to create socket {}{}", addr->sun_path[0] ? addr->sun_path[0] : '@', addr->sun_path + 1);
+        Debug::log(ERR, "Failed to create socket {}{}", addr->sun_path[0] ? addr->sun_path[0] : '@', addr->sun_path + 1);
         return -1;
     }
-    if (!set_cloexec(fd, true)) {
+
+    if (!setCloseOnExec(fd, true)) {
         close(fd);
         return -1;
     }
 
     if (addr->sun_path[0]) {
         unlink(addr->sun_path);
     }
+
     if (bind(fd, (struct sockaddr*)addr, size) < 0) {
-        rc = errno;
-        Debug::log(ERR, "failed to bind socket {}{}", addr->sun_path[0] ? addr->sun_path[0] : '@', addr->sun_path + 1);
-        goto cleanup;
+        Debug::log(ERR, "Failed to bind socket {}{}", addr->sun_path[0] ? addr->sun_path[0] : '@', addr->sun_path + 1);
+        cleanUpSocket(fd, addr->sun_path);
+        return -1;
     }
-    if (listen(fd, 1) < 0) {
-        rc = errno;
-        Debug::log(ERR, "failed to listen to socket {}{}", addr->sun_path[0] ? addr->sun_path[0] : '@', addr->sun_path + 1);
-        goto cleanup;
+
+    if (listen(fd, SOCKET_BACKLOG) < 0) {
+        Debug::log(ERR, "Failed to listen to socket {}{}", addr->sun_path[0] ? addr->sun_path[0] : '@', addr->sun_path + 1);
+        cleanUpSocket(fd, addr->sun_path);
+        return -1;
     }
 
     return fd;
-
-cleanup:
-    close(fd);
-    if (addr->sun_path[0]) {
-        unlink(addr->sun_path);
-    }
-    errno = rc;
-    return -1;
 }
 
 static bool checkPermissionsForSocketDir(void) {
     struct stat buf;
 
     if (lstat("/tmp/.X11-unix", &buf)) {
-        Debug::log(ERR, "Failed statting X11 socket dir");
+        Debug::log(ERR, "Failed to stat X11 socket dir");
         return false;
     }
 
     if (!(buf.st_mode & S_IFDIR)) {
-        Debug::log(ERR, "X11 socket dir is not a dir");
+        Debug::log(ERR, "X11 socket dir is not a directory");
         return false;
     }
 
     if (!((buf.st_uid == 0) || (buf.st_uid == getuid()))) {
-        Debug::log(ERR, "X11 socket dir is not ours");
+        Debug::log(ERR, "X11 socket dir is not owned by root or current user");
         return false;
     }
 
     if (!(buf.st_mode & S_ISVTX)) {
         if ((buf.st_mode & (S_IWGRP | S_IWOTH))) {
-            Debug::log(ERR, "X11 socket dir is sticky by others");
+            Debug::log(ERR, "X11 socket dir is writable by others");
             return false;
         }
     }
 
     return true;
 }
 
-static bool openSockets(std::array<int, 2>& sockets, int display) {
-    auto ret = mkdir("/tmp/.X11-unix", 755);
-
-    if (ret != 0) {
-        if (errno == EEXIST) {
-            if (!checkPermissionsForSocketDir())
-                return false;
+static bool ensureSocketDirExists(){
+    if(mkdir("/tmp/.X11-unix", SOCKET_DIR_PERMISSIONS)!=0){
+        if(errno == EEXIST){
```

Comment:
no {}

Reply:
I left this one. I was getting a dangling else warning if I removed the {}

Reply:
the `errno` one, not the parent

Reply:
done